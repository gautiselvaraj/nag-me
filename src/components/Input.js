import React from 'react';
import styled from 'styled-components';
import Label from './Label';

const InputWrap = styled.div`
  display: flex;
  flex-direction: ${props => props.inline ? 'row' : 'column'};
`;

const InputError = styled.div`
  color: red;
  font-size: .85rem;
  margin-top: 5px;
`;

const Input = styled.input`
  border: 1px solid ${props => props.error ? 'red' : props.theme.greyLighter};
  border-radius: ${props => props.search ? '1rem' : '5px'};
  box-sizing: border-box;
  font-size: ${props => props.small ? '.85rem' : '1rem'};
  height: ${props => props.small ? '1.5rem' : '2rem'};
  padding: 0 .35rem;
  ${props => props.inline ? '' : 'width: 100%;'};
`;

export default ({label, error, id, ...otherProps}) => (
  <InputWrap {...otherProps}>
    {!!label &&
      <Label htmlFor={id} {...otherProps}>{label}</Label>
    }
    <Input id={id} error={error} {...otherProps} />
    {!!error &&
      <InputError>{error}</InputError>
    }
  </InputWrap>
)
