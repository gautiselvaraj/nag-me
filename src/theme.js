export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
