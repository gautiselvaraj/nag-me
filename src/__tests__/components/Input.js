import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Input, { InputError } from '../../components/Input';
import theme from '../../theme';
import 'jest-styled-components';

describe('<Input />', () => {
  describe('without label and error', () => {
    it('should match snapshot', () => {
      const tree = renderer.create(<Input theme={theme} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have correct element height & font-size based on small prop', () => {
      let input = mount(<Input theme={theme} />);
      expect(input.find('input').length).toBe(1);
      expect(input.find('input')).toHaveStyleRule('font-size', '1rem');
      expect(input.find('input')).toHaveStyleRule('height', '2rem');

      input = mount(<Input small theme={theme} />);
      expect(input.find('input')).toHaveStyleRule('font-size', '.85rem');
      expect(input.find('input')).toHaveStyleRule('height', '1.5rem');
    });
  });

  describe('with label and error', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(<Input label="Label 1" error="Error 1" theme={theme} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should have label and error element if passed', () => {
      let input = mount(
        <Input label="Label 1" error="Error 1" theme={theme} />
      );
      expect(input.find('label').length).toBe(1);
      expect(input.find('label').text()).toBe('Label 1');
      expect(input.find(InputError).text()).toBe('Error 1');
      expect(input.find(InputError)).toHaveStyleRule('color', 'red');
    });
  });
});
