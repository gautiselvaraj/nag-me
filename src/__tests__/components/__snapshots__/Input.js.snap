// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> with label and error should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  border: 1px solid red;
  border-radius: 5px;
  box-sizing: border-box;
  font-size: 1rem;
  height: 2rem;
  padding: 0 0.35rem;
  width: 100%;
}

.c1 {
  font-size: 1rem;
  line-height: 2rem;
}

.c3 {
  color: red;
  font-size: 0.85rem;
  margin-top: 5px;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    Label 1
  </label>
  <input
    className="c2"
    id={undefined}
  />
  <div
    className="c3"
  >
    Error 1
  </div>
</div>
`;

exports[`<Input /> without label and error should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  border: 1px solid #dddddd;
  border-radius: 5px;
  box-sizing: border-box;
  font-size: 1rem;
  height: 2rem;
  padding: 0 0.35rem;
  width: 100%;
}

<div
  className="c0"
>
  <input
    className="c1"
    id={undefined}
  />
</div>
`;
